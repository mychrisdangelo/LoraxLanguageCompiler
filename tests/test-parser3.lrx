int main() {
	bool b;
	a = b = c;
	z%0%3@ = 4;
	a = z%0;
	t%0 = t2; // t and t2 are both trees. t2 is now being assigned the first child of t (accessed by %0)
	t%0 = t3%0; // similar to above but now t's first child is t3's first child.
	t%0@ = 3; // @ dereferences the value in that node. Now the t's first child's value is 3
	t%3+4@ = 4;
	normal_int = t%0@; // now we're assigning a normal int var the value from inside t's first child value
	t@ = 4; // this is assigning the root nodes value as 4
	t%0%1@ = 5; // this is assigning t's first child's, second child node value to 5
	t3 = t%0%1--; // this is popping t's first child's second child node from the tree t and returns t to assign to t3
	t%toyfunc()%3@-- = t2--;
	t = 5[];
	b = (a%0 == null);
}