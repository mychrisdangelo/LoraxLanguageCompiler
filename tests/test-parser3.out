int main()
{
bool b;
a = b = c;
z % 0 % 3@ = 4;
a = z % 0;
t % 0 = t2;
t % 0 = t3 % 0;
t % 0@ = 3;
t % 3 + 4@ = 4;
normal_int = t % 0@;
t@ = 4;
t % 0 % 1@ = 5;
t3 = t % 0 % 1--;
t % toyfunc() % 3@-- = t2--;
t = 5[];
b = a % 0 == null;
}